### ISSUE TITLE
Missing the "Fully check the document" command of CoqIDE

### ISSUE THREAD

**Issue by MSoegtropIMC** [coq-community/vscoq#287]
When parallel proof mode is enabled, proofs are done in arbitrary order **assuming** that upstream proofs can be proven. CoqIDE has a button "Fully check the document" which eventually ensures that all these assumptions hold. The `Qed` lines change color in this step in CoqIDE. I lack this button (or command) in VsCoq.

**Comment by artagnon** [comment]
I've never used CoqIDE, so I might be wrong about this, but is "extension.coq.interpretToEnd" the command you're looking for?

**Comment by MSoegtropIMC** [comment]
No. It is neither "interpret to end" nor "synchronous interpret to end". There is a missing piece which lifts an asynchronous interpret to end to a synchronous interpret to end.

@gares : maybe you can give some details on this.

**Comment by gares** [comment]
No, it is not about jumping to the end. The XML protocol has a special call to wait for all workers to finish their job and check in the master process that the proof they built is OK (they do it locally as well, but then don't have the "global picture of universes & co").

Given that the current vscoq uses behind the scenes the XML protocol, this can be implemented.

In the new LSP based vscoqtop I've been working on with Maxime, this thing is not there anymore. In the sense that workers don't send back the proof at all. So in interactive mode you trust they check it, then you coqc the file and you do the "global check".

The original design was inspired by Isabelle, where "async proofs" are part of the logic in some sense. I think this did not work well, and made many things more complicated than needed. The new design (thanks @ppedrot for doing most of the work, https://github.com/coq/ceps/pull/40) sees "async proofs" as an artifact of the UI, it is not the logic (kernel) but rather the UI (the document manager) which keeps track of these delegated proofs. There is still code to check proofs, coqc uses it, but I don't think hooking it up in the UI adds a huge benefit.

**Comment by MSoegtropIMC** [comment]
So you say eventually it will go away - I guess we don't need it in VsCoq then.

But then I then don't understand is that in VsCoq the Qeds are marked in some violet - I assumed that this matches the CoqIDE marking of a "preliminary not yet globally checked Qed".  So I think either there should be a function to do the final check or this marking of Qeds should go away (it is actually quite annoying because unlike in CoqIDE it occupies a minimum size of several pixels in the side bar and obscures e.g. the progress of the parallel proofs - the blue is hard to see among the violet of Qed.

Maybe this is also a misunderstanding on my side and VsCoq always marks Qeds like this. But then I find the coloring / pixel size in the bar rather questionable.

**Comment by Zimmi48** [comment]
I've also had confused students ask me about this violet and I just told them they should ignore it because it was related to some universe stuff that they didn't need to understand...

**Comment by MSoegtropIMC** [comment]
@rtetley : I am not happy with closing this. @gares said it will eventually go away, but then the colouring is confusing.

**Comment by rtetley** [comment]
@MSoegtropIMC That's fair, I'll reopen it so long. Can we maybe make a separate issue for the colouring ? Also I'm not certain the violet colouring still happens with VsCoq 2. I think there might be some tweeking to do with regards to the decorations.

**Comment by MSoegtropIMC** [comment]
I would say only @gares can make a final statement on what makes sense in which version of Coq / VsCoq. All I can say is that the status quo is confusing and not helpful, because the violet markings hide interesting information.
